src/main/groovy/org/boozallen/plugins/jte/Utils.groovy

```diff
@@ -145,10 +145,59 @@ class Utils implements Serializable{
         ItemGroup<?> parent = job.getParent()
         PrintStream logger = getLogger() 
 
-        if (!scm) scm = getSCMFromJob(job, parent)
+        // create SCMFileSystem 
+        SCMFileSystem fs = null
+
+        // if provided a SCM, try to build directly: 
+        if (scm){
+            if (SCMFileSystem.supports(scm)){
+                fs = SCMFileSystem.of(job, scm) 
+            }else{
+                return null 
+            }
+        }else{ // try to infer SCM info from job properties 
+            
+            // for multibranch projects, need to determine SCM from parent 
+            // and branch specific info from job 
+            if (parent instanceof WorkflowMultiBranchProject){
+                // ensure branch is defined 
+                BranchJobProperty property = job.getProperty(BranchJobProperty.class)
+                if (!property){
+                    throw new IllegalStateException("inappropriate context")
+                }
+
+                Branch branch = property.getBranch()
+
+                // get scm source for specific branch and ensure present
+                // (might not be if branch deleted after job triggered)
+                String branchName = branch.getSourceId()
+                SCMSource scmSource = parent.getSCMSource(branchName)
+                if (!scmSource) {
+                    throw new IllegalStateException("${branch.getSourceId()} not found")
+                }
+
+                // attempt lightweight checkout
+                /*
+                    some hacky stuff here.. can't make GitSCMFileSystem from a PR.. so if PR -> use source branch
+                */ 
+                SCMHead head = branch.getHead()
+                if (head instanceof PullRequestSCMHead){
+                    head = new BranchSCMHead(head.sourceBranch)
+                }
+                
+                fs = SCMFileSystem.of(scmSource, head) 
+
+            } else { // then just a regular pipeline job: 
+                FlowDefinition definition = job.getDefinition() 
```

Comment:
should we report something or is it ok if this block's 'if' conditions are not met?